### DISCUSSION TITLE
How do I delete run record completely including the one in the tracking table?

### DISCUSSION THREAD

**Discussion by yitistica** [mlflow/mlflow#7618] [Q&A] [answered]
When I execute the following:

```python
client = MlflowClient()
client.delete_run(run_id)
```
it seems to only transit the lifecycle stage of a run to `deleted`, but it doesn't delete the associated artifacts nor remove the record row from the `runs` table used for tracking. Is there a way in the Mlfow API to actually wipe clean everything related to a run (including removing all the artifacts under the run and deleting all the related records in the tracking tables? 

The reason is that my application is expected to generate thousands of runs daily, it may cause some speed issues as these records pile up in the table.

**Comment by JHibbard** [comment, marked as answer]
Hi yitistica,

To permanently delete run data, you can use the [gc cli utility](https://www.mlflow.org/docs/latest/cli.html#mlflow-gc). To permanently delete experiments, you'll have to use [mlflow experiments delete](https://www.mlflow.org/docs/latest/cli.html#mlflow-experiments-delete).

Hope that helps!
Jim
